api/lib/opentelemetry/context.rb

```diff
@@ -75,31 +96,26 @@ def with_value(key, value)
       #   to the block
       def with_values(values)
         ctx = current.set_values(values)
-        prev = ctx.attach
+        prev = attach(ctx)
         yield ctx, values
       ensure
-        ctx.detach(prev)
-      end
-
-      # Returns the value associated with key in the current context
-      #
-      # @param [String] key The lookup key
-      def value(key)
-        current.value(key)
+        detach(prev)
       end
 
       def clear
         self.current = ROOT
       end
 
       def empty
-        new(nil, EMPTY_ENTRIES)
+        new(EMPTY_ENTRIES)
       end
     end
 
-    def initialize(parent, entries)
-      @parent = parent
+    attr_accessor :parent
+
+    def initialize(entries)
       @entries = entries.freeze
+      @parent = parent
```

Comment:
🤔 I am very confused - won't `parent` always be `nil` here?

Reply:
![](https://media.tenor.com/images/1fb93e098589d31f5023b7d32418f013/tenor.gif)

Reply:
Much more likely a bug though.  I expect this to change based on the comments above.